src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -177,6 +177,12 @@ private function parseDefinition($id, $service, $file)
             $public = !array_key_exists('public', $service) || (bool) $service['public'];
             $this->container->setAlias($id, new Alias($service['alias'], $public));
 
+            foreach ($service as $key => $value) {
+                if ($key !== 'alias' && $key !== 'public') {
```

Comment:
Sure I'll change it this afternoon. But won't we suffer performance issues with ``in_array``?